lib/mongoid/stateful.rb

```diff
@@ -31,6 +48,15 @@ def persisted?
       !new_record? && !destroyed?
     end
 
+    # Checks if the document was previously saved to the database
+    # but now it has been deleted.
+    #
+    # @return [ true, false ] True if was persisted but now destroyed,
+    #   otherwise false.
+    def previously_persisted?
+      !new_record? && destroyed?
```

Comment:
The description of this method should be:

```
    # Returns true unless the document was just created -- that is, the object already persisted prior to its last save.
```

I believe the logic should be:

````
  !previously_new_record?
```

I'm pretty sure we don't need to check for `destroyed?` here. The fact that the object was destroyed on the current action shouldn't influence its *previous* state.

Reply:
As far as I can tell the current docstring matches what is described in https://sapidlabs.com/rails/2021/08/28/rails-7-adds-previously-persisted.html. @johnnyshields are you suggesting an alternate way of communicating the behavior described in this post for rails? Seems to me that your suggestion is to have different behavior.

Reply:
You're right, please disregard my comment. OK, makes sense to follow the Rails way.